airbyte-cdk/python/unit_tests/sources/declarative/async_job/test_job_orchestrator.py

```diff
@@ -140,5 +140,41 @@ def test_when_fetch_records_then_yield_records_from_each_job(self) -> None:
         assert len(records) == 2
         assert self._job_repository.fetch_records.mock_calls == [call(first_job), call(second_job)]
 
+    @mock.patch(sleep_mock_target)
+    def test_given_exception_when_start_job_and_raise_this_exception(self, mock_sleep: MagicMock) -> None:
```

Comment:
There is one thing I would like to improve on this test and it is that it relies on private attributes/methods like `_start_jobs` and `_running_partitions`. Could we update this to avoid using those. Something like this:
```
    @mock.patch(sleep_mock_target)
    def test_given_exception_when_start_job_and_raise_this_exception_and_abort_jobs(self, mock_sleep: MagicMock) -> None:
        orchestrator = AsyncJobOrchestrator(self._job_repository, [_A_STREAM_SLICE, _ANOTHER_STREAM_SLICE], exceptions_to_break_on=[ValueError])
        self._job_repository.start.side_effect = [self._job_for_a_slice, ValueError("Something went wrong")]

        with pytest.raises(ValueError):
            # assert that orchestrator exits on expected error
            list(orchestrator.create_and_get_completed_partitions())
        assert len(orchestrator._running_partitions) == 0
        self._job_repository.abort.assert_called_once_with(self._job_for_a_slice)
```

Reply:
updated